hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/TestHoodieSparkSqlWriter.scala

```diff
@@ -454,6 +454,38 @@ class TestHoodieSparkSqlWriter {
     assert(df.except(trimmedDf).count() == 0)
   }
 
+  /**
+   * Test case for upsert dataset without precombine field and with
+   * auto adjusted COMBINE_BEFORE_UPSERT set to false.
+   */
+  @Disabled
```

Comment:
can be disabled after https://github.com/apache/hudi/pull/7998 is merged and this branch is rebased on it

Reply:
I should probably also add E2E test for SQL update